### PR TITLE
CAMEL-14154: Create default cacheContainer if no configuration is provided

### PR DESCRIPTION
**Pull request by mmelko**
CAMEL-14154: Create default cacheContainer if no configuration is provided
Add tests for embedded chache
Mention embedded cacheContainer in doc

### CODE REVIEW

File: components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java [hunk 1]
```diff
@@ -69,7 +74,20 @@ protected Endpoint createEndpoint(String uri, String remaining, Map<String, Obje
         } else {
             conf = new InfinispanConfiguration();
         }
-        conf.setCacheContainer(cacheContainer);
+
+        String[] confParameters = new String[] {"hosts", "cacheContainerConfiguration", "configurationUri", "cacheContainer"};
+        //     init default embedded cache if config parameters aren't specified or cacheContainer is set using setMethod
+        if (!Arrays.stream(confParameters).anyMatch(parameters::containsKey) || setCacheFromComponent) {
```
Comment (lburgazzoli) [inline components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java]: you probably need to check also the values set on the `InfinispanConfiguration`

Comment (oscerd) [reply components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java]: Missed this one.

Comment (oscerd) [reply components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java]: Thanks @lburgazzoli

Comment (davsclaus) [reply components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java]: Is the `||` OR correct as the `setCacheFromComponent` is always true. So I think its is wrong, and it should be `&&` AND instead

**Comment by davsclaus** [comment]
Also maybe add a DEBUG log or something that a default cache is created/will be used

File: components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java (hunk 1 above)
Comment (mmelko) [reply components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java]: thanks. I didn't know about other possibilities of initialization. I added help method do determine whether configuration is provided or not.

Comment (mmelko) [reply components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java]: @davsclaus actually `setCacheFromComponent` is almost never true, at least in tests which I provided. Usually cacheContainer is initialized in `InfinispanManager` class.

File: components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java [hunk 2]
```diff
@@ -59,6 +63,8 @@ public BasicCacheContainer getCacheContainer() {
      */
     public void setCacheContainer(BasicCacheContainer cacheContainer) {
         this.cacheContainer = cacheContainer;
+        this.setCacheFromComponent = true;
+        System.out.println("===========<created");
```
Comment (davsclaus) [inline components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/InfinispanComponent.java]: Remove

**Comment by davsclaus** [comment]
There is some system out println to be removed

**Comment by mmelko** [comment]
> There is some system out println to be removed

sorry for that :) It's gone.

**Comment by mmelko** [comment]
@oscerd hmm, that's weird.. tests are passing on my local.

**Comment by oscerd** [comment]
No idea. Still failing for me.

**Comment by oscerd** [comment]
Merged on master

**Comment by oscerd** [comment]
Thanks

**Pull request closed without merging**